service/topology_coordinator.cc

```diff
@@ -1423,8 +1423,14 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
                     }
 
                     if (action_failed(tablet_state.streaming)) {
-                        bool cleanup = utils::get_local_injector().enter("stream_tablet_move_to_cleanup");
-                        if (cleanup || check_excluded_replicas()) {
+                        const bool cleanup = utils::get_local_injector().enter("stream_tablet_move_to_cleanup");
+                        const bool critical_disk_utilization = _tablet_allocator.get_load_stats()->critical_disk_utilization
+                                .or_else([&host=trinfo.pending_replica->host] {
+                                    return std::optional(decltype(locator::load_stats::critical_disk_utilization)::value_type({{host, false}}));
+                                })
+                                .value()[trinfo.pending_replica->host];
```

Comment:
Here too I think that having critical_disk_utilization non-optional wold simplify the code, e.g.
```c++
const auto& util = _tablet_allocator.get_load_stats()->critical_disk_utilization;
auto it = util.find(trinfo.pending_replica->host);
bool critical_disk_utilization = it != util.end() && it->second;
```